views/default/input/date.php

```diff
@@ -35,19 +34,30 @@
 $timestamp = $vars['timestamp'];
 unset($vars['timestamp']);
 
-if ($timestamp) {
-	echo elgg_view('input/hidden', ['name' => $vars['name'], 'value' => $vars['value']]);
+// convert timestamps to text for display
+$ts_value = '';
+if (isset($vars['value'])) {
+	if (is_numeric($vars['value'])) {
+		$ts_value = $vars['value'];
+		$vars['value'] = gmdate('Y-m-d', $vars['value']);
+	} else {
+		$ts_value = strtotime($vars['value']);
```

Comment:
It's not opinion. With this change plugin devs can reliably pass in strings like "+1 week" and they will be turned into working timestamps. And that usage would fail on previous 1.12.x versions.